botocore/data/ec2/2015-10-01/waiters-2.json

```diff
@@ -7,9 +7,8 @@
       "operation": "DescribeInstances",
       "acceptors": [
         {
-          "matcher": "path",
-          "expected": true,
-          "argument": "length(Reservations[]) > `0`",
```

Comment:
I believe this was added because it's possible to get an empty reservations list even if you specify an instance id in some cases.  I think we'll just need to modify the waiter code to handle this case appropriately.